drivers/flash/soc_flash_mcux.c

```diff
@@ -36,16 +36,16 @@ LOG_MODULE_REGISTER(flash_mcux);
 #error No matching compatible for soc_flash_mcux.c
 #endif
 
-#ifdef SOC_HAS_IAP
+#if defined(SOC_HAS_IAP) && !defined(CONFIG_SOC_LPC55S36)
```

Comment:
To clarify here, I am assuming the reason that this change is present is because the LPC55S36 uses the `fsl_flash.h` header file? If so, I'd prefer we change the `SOC_HAS_IAP` define to something like `SOC_USES_IAP_API`, and then left that option undefined for the LPC55S36, rather than introducing this SOC specific dependency in the driver.